### PR TITLE
Fix PrestoQueryRunner::toSql to honor sortingKeys for Aggregates.

### PR DESCRIPTION
**Pull request by kgpai**
Currently PrestoQueryRunner doesnt honor sortingKeys if its an aggregation plan node. This is required to fix mismatch when running the AggregationFuzzer with --enableSortedInput and using Presto as source of truth.

### CODE REVIEW

File: velox/exec/fuzzer/PrestoQueryRunner.cpp [hunk 1]
```diff
@@ -240,11 +244,23 @@ std::string toCallSql(const core::CallTypedExprPtr& call) {
...
 
-      sql << ") -> " << toCallSql(body);
+      sql << ") -> " << toCallSql(body, {}, {});
     } else {
       VELOX_NYI();
     }
   }
+
+  if (!sortingKeys.empty()) {
+    VELOX_CHECK_EQ(sortingKeys.size(), sortingOrder.size());
```
Comment (duanmeng) [inline velox/exec/fuzzer/PrestoQueryRunner.cpp:254]: Should we move this check to the beginning of this function as their size should always be the same?

File: velox/exec/tests/PrestoQueryRunnerTest.cpp [hunk 2]
```diff
@@ -95,4 +99,44 @@ TEST_F(PrestoQueryRunnerTest, DISABLED_fuzzer) {
   velox::exec::test::assertEqualResults(
       prestoResults, plan->outputType(), {veloxResults});
 }
+
+TEST_F(PrestoQueryRunnerTest, sortedOrder) {
```
Comment (duanmeng) [inline velox/exec/tests/PrestoQueryRunnerTest.cpp:103]: Should we disable this test by default?

File: velox/exec/fuzzer/PrestoQueryRunner.cpp [hunk 3]
```diff
@@ -291,7 +307,8 @@ std::optional<std::string> PrestoQueryRunner::toSql(
     for (auto i = 0; i < aggregates.size(); ++i) {
       appendComma(i, sql);
       const auto& aggregate = aggregates[i];
-      sql << toCallSql(aggregate.call);
+      sql << toCallSql(
+          aggregate.call, aggregate.sortingKeys, aggregate.sortingOrders);
```
Comment (mbasmanova) [inline velox/exec/fuzzer/PrestoQueryRunner.cpp:311]: perhaps, add a check that aggregate.distinct is false; otherwise, we may end up silently generating invalid SQL

File: velox/exec/tests/PrestoQueryRunnerTest.cpp (hunk 2 above)
Comment (kgpai) [reply velox/exec/tests/PrestoQueryRunnerTest.cpp:103]: I kept it enabled since I was able to run this without a local presto instance listening on 8080. It doesnt seem like we eagerly establish a connection when we initialize PQR, should that change this will fail. I can still DISABLE it if you insist though.

File: velox/exec/fuzzer/PrestoQueryRunner.cpp [hunk 4]
```diff
@@ -211,7 +212,11 @@ std::string toTypeSql(const TypePtr& type) {
   }
 }
 
-std::string toCallSql(const core::CallTypedExprPtr& call) {
+std::string toCallSql(
+    const core::CallTypedExprPtr& call,
+    const std::vector<core::FieldAccessTypedExprPtr>& sortingKeys,
+    const std::vector<core::SortOrder>& sortingOrder) {
```
Comment (mbasmanova) [inline velox/exec/fuzzer/PrestoQueryRunner.cpp:218]: naming: sortingOrder -> sortingOrders to match sortingKeys

File: velox/exec/fuzzer/PrestoQueryRunner.cpp [hunk 5]
```diff
@@ -291,7 +306,11 @@ std::optional<std::string> PrestoQueryRunner::toSql(
     for (auto i = 0; i < aggregates.size(); ++i) {
       appendComma(i, sql);
       const auto& aggregate = aggregates[i];
-      sql << toCallSql(aggregate.call);
+      VELOX_CHECK(
+          !aggregate.distinct,
+          "Presto Query Runner doesnt support distinct aggregates");
```
Comment (mbasmanova) [inline velox/exec/fuzzer/PrestoQueryRunner.cpp:311]: typo: doesnt -> doesn't or does not

Would be nice to remote this limitation in a follow-up. Should be straightforward.

File: velox/exec/tests/PrestoQueryRunnerTest.cpp [hunk 6]
```diff
@@ -32,6 +32,10 @@ namespace facebook::velox::exec::test {
 class PrestoQueryRunnerTest : public ::testing::Test,
                               public velox::test::VectorTestBase {
  protected:
+  static void SetUpTestCase() {
+    memory::MemoryManager::testingSetInstance({});
```
Comment (mbasmanova) [inline velox/exec/tests/PrestoQueryRunnerTest.cpp:36]: Why is this needed? CC: @xiaoxmeng

File: velox/exec/tests/PrestoQueryRunnerTest.cpp [hunk 7]
```diff
@@ -95,4 +99,52 @@ TEST_F(PrestoQueryRunnerTest, DISABLED_fuzzer) {
   velox::exec::test::assertEqualResults(
       prestoResults, plan->outputType(), {veloxResults});
 }
+
+TEST_F(PrestoQueryRunnerTest, sortedOrder) {
```
Comment (mbasmanova) [inline velox/exec/tests/PrestoQueryRunnerTest.cpp:103]: naming: sortedOrder -> sortedAggregation

File: velox/exec/tests/PrestoQueryRunnerTest.cpp [hunk 8]
```diff
@@ -95,4 +99,52 @@ TEST_F(PrestoQueryRunnerTest, DISABLED_fuzzer) {
...
+  auto queryRunner =
+      std::make_unique<PrestoQueryRunner>("http://127.0.0.1:8080", "hive");
```
Comment (mbasmanova) [inline velox/exec/tests/PrestoQueryRunnerTest.cpp:105]: perhaps, change "http://127.0.0.1:8080" to "http://unused" to make it clear that this test doesn't require running Presto instance.

File: velox/exec/tests/PrestoQueryRunnerTest.cpp [hunk 9]
```diff
@@ -95,4 +99,52 @@ TEST_F(PrestoQueryRunnerTest, DISABLED_fuzzer) {
...
+          .planNode();
+
+  auto sql = queryRunner->toSql(plan);
+  ASSERT_TRUE(sql.has_value());
+
+  ASSERT_EQ(
+      "SELECT multimap_agg(c0, c1 ORDER BY c0 ASC NULLS LAST) as a0 FROM tmp",
+      sql.value());
+
+  // multiple order by's
```
Comment (mbasmanova) [inline velox/exec/tests/PrestoQueryRunnerTest.cpp:126]: Make sure comments are full sentences.

File: velox/exec/tests/PrestoQueryRunnerTest.cpp [hunk 10]
```diff
@@ -95,4 +99,52 @@ TEST_F(PrestoQueryRunnerTest, DISABLED_fuzzer) {
...
+              {"multimap_agg(c0, c1 order by c1 asc nulls first, c0 desc nulls last, c2 asc nulls last)"})
+          .planNode();
+
+  sql = queryRunner->toSql(plan);
+  ASSERT_TRUE(sql.has_value());
+  ASSERT_EQ(
+      "SELECT multimap_agg(c0, c1 ORDER BY c1 ASC NULLS FIRST, c0 DESC NULLS LAST, c2 ASC NULLS LAST) as a0 FROM tmp",
+      sql.value());
+
+  plan = velox::exec::test::PlanBuilder()
```
Comment (mbasmanova) [inline velox/exec/tests/PrestoQueryRunnerTest.cpp:142]: Please, extract this into a separate test case.

File: velox/exec/tests/PrestoQueryRunnerTest.cpp [hunk 11]
```diff
@@ -95,4 +99,52 @@ TEST_F(PrestoQueryRunnerTest, DISABLED_fuzzer) {
...
+             .values({data})
+             .singleAggregation(
+                 {}, {"multimap_agg(distinct c0, c1 order by c0 asc)"})
+             .planNode();
+
+  ASSERT_THROW(queryRunner->toSql(plan), VeloxException);
```
Comment (mbasmanova) [inline velox/exec/tests/PrestoQueryRunnerTest.cpp:148]: use VELOX_ASSERT_THROW

**Review by mbasmanova — COMMENTED** [review]
@kgpai Looks good to me % a few nits.

File: velox/exec/fuzzer/PrestoQueryRunner.cpp (hunk 5 above)
Comment (kgpai) [reply velox/exec/fuzzer/PrestoQueryRunner.cpp:311]: Will remove in follow up.

File: velox/exec/tests/PrestoQueryRunnerTest.cpp (hunk 6 above)
Comment (kgpai) [reply velox/exec/tests/PrestoQueryRunnerTest.cpp:36]: The test fails claiming memory manager not initialized without this.
